docs/metrics/index.md

```diff
@@ -1,269 +1,140 @@
-application for further aggregation. These aggregations happen within the agent. 
-
-Look [here](statsd.md) for a more detailled description of the ZMX StatsD support.
-
----
-**NOTE**
-
-There is one exception in ZIO ZMX per design choice: We do allow gauges to change relative to the previous value. Therefore we are keeping 
-track of the last values recorded for all of the Gauges also for the statsd backend.
-
----
-
-### Prometheus 
-
-Within a typical prometheus environment we also find at least one component collecting the data, the prometheus server. The prometheus 
-server provides more functionality than the statsd collector. It organizes its data, so that it can be queried using the [_Prometheus 
-Query Language_](https://prometheus.io/docs/prometheus/latest/querying/basics/). 
-
-However, a typical setup is that one or more Prometheus servers are configured as a datasource within a [Grafana](https://prometheus.io/docs/prometheus/latest/querying/basics/) server for further aggregation and visualization. 
-
-A single prometheus server normally collects its data from configured (HTTP) endpoints, each of which provides the collected metrics in a 
-special [text format](https://prometheus.io/docs/instrumenting/exposition_formats/). There are many exporters within the prometheus 
-ecosystem, but most of them map some application specific metrics to a prometheus compatible endpoint via HTTP. 
-
---- 
-**NOTE**
-
-ZMX provides the Prometheus data using the `report` effect in the `Instrumentation`trait, but does not make any assumptions how that 
-data shall be offered via HTTP. Users have to use the HTTP stack of their choice to provide the HTTP endpoint to Prometheus. 
-
----
-
-Within prometheus the application is responsible for keeping track of the collected metrics and provide the data to prometheus upon request. 
-
-As a consequence, the Prometheus implementation uses a data registry underneath, which is modified by the stream of `MetricEvent`s and 
-provides the input to generate the prometheus data. 
+```scala
+def countValue(name: String, tags: Label*): MetricAspect[Double]
+```
 
-Look [here](prometheus.md) for a more detailled description of the ZMX Prometheus support.
+A counter that can be applied to effects having the result type `A`. Given the effect 
+produces `v: A`, the counter will be increased by `f(v)`.
+```scala
+def countValueWith[A](name: String, tags: Label*)(f: A => Double): MetricAspect[A]
+```
 
-## Metrics supported in ZMX
+### Examples
 
-### Counter 
+Create a counter named `countAll` which is incremented by `1` every time it is invoked. 
 
-A `Counter`in ZMX is a monotonically increasing series of values. Even though statsd and Datadog would allow counters to decrease, 
-we enforce that counters in ZMX can only increase. As a consequence, a metric that can either increase or decrease should be a gauge. 
+```scala mdoc:silent
+val aspCountAll = MetricAspect.count("countAll")
```

Comment:
Maybe just `countAll` instead of `aspCountAll` and similarly for other metric aspects?